src/redux/create-module.js

```diff
@@ -35,33 +35,40 @@ function createModule(namespace, createDefinition) {
   // ----------------------------------
   // Ensure that the generated definition conforms to the required interface.
   invariant(
-    typeof definition === 'object',
+    objectClass(definition) === 'object',
     `Error in \`createModule\` for the "${namespace}" namespace. The value ` +
     'returned from \`createDefinition\` must be an object. Instead, what was ' +
     'received was of type: %s.',
-    typeof definition
+    objectClass(definition)
   )
   invariant(
-    definition.initialState != null, // eslint-disable-line
+    definition.initialState !== null && definition.initialState !== undefined,
     `Error in \`createModule\` for the "${namespace}" namespace. The object ` +
     'returned from \`createDefinition\` must define an \`initialState\` ' +
     'property that is a not a null or undefined value. Instead, what was ' +
     'received was of type: %s.',
-    typeof definition.initialState
+    objectClass(definition.initialState)
   )
   invariant(
-    typeof definition.events === 'object',
+    objectClass(definition.events) === 'object',
     `Error in \`createModule\` for the "${namespace}" namespace. The object ` +
     'returned from \`createDefinition\` must define an \`events\` property ' +
     'that is an object. Instead, what was received was of type: %s.',
-    typeof definition.events
+    objectClass(definition.events)
   )
   invariant(
-    typeof definition.handlers === 'object',
+    objectClass(definition.handlers) === 'object',
     `Error in \`createModule\` for the "${namespace}" namespace. The object ` +
     'returned from \`createDefinition\` must define a \`handlers\` property ' +
     'that is an object. Instead, what was received was of type: %s.',
-    typeof definition.handlers
+    objectClass(definition.handlers)
+  )
+  invariant(
+    objectClass(definition.selectors) === 'object',
```

Comment:
We now require selectors as well, I wasn't 100% on this at first but it seems like a bad practice to get state any other way.  Example, in `connect()` you don't want to be manually traversing state when you should be using a selector.  I think this warrants any module that is dealing with state, should always provide selectors for that state.